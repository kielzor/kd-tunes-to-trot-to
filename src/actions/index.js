export const addUserFile = audioClip => {
  return {
    type: 'ADD_NEW_FILE',
    audioClip
  };
};

export const storeJog = audioClip => {
  return {
    type: 'ADD_JOG_TRACKS',
    audioClip
  };
};

export const storeWarmup = audioClip => {
  return {
    type: 'ADD_WARMUP_TRACKS',
    audioClip
  };
};

export const storeSprint = audioClip => {
  return {
    type: 'ADD_SPRINT_TRACKS',
    audioClip
  };
};

export const storeWorkout = workout => {
  return {
    type: 'CREATE_WORKOUT',
    workout
  };
};

export const populateWorkout = workout => {
  return {
    type: 'POPULATE_WORKOUT',
    workout
  };
};