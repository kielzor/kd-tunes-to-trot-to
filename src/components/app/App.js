import React, { Component } from 'react';
import './App.css';
import { clientId, secretId } from '../../keys'
import Header from '../header/header'
import DesktopDisplayContainer from '../desktopDisplayContainer/DesktopDisplayContainer.js'
import { Route, withRouter, Link } from 'react-router-dom';
import { ImportContainer } from '../importContainer/ImportContainer'
import { NewSongContainer } from '../newSongContainer/NewSongContainer'
import { WorkoutContainer } from '../workoutContainer/WorkoutContainer'
// import { connect } from 'react-redux'


class App extends Component {
  constructor() {
    super()
    this.state = {
      desktopClicked: false,
      code: ''
    }
  }
  
  componentDidMount() {
    this.setState({code: window.location.search.split('').splice(6,window.location.search.length).join('')})
  }

  handleDesktopClick = (type) => {
    if (type === 'back') this.setState({ desktopClicked: false})
    else this.setState ({ desktopClicked: true })
  }

  logOut = () => {
    this.setState({
      code: ''
    })
    this.handleDesktopClick('back')
  }

  handleSubmit = async () => {
    try {
      const response = await fetch(`https://freesound.org/apiv2/oauth2/access_token?client_id=${clientId}&client_secret=${secretId}&grant_type=authorization_code&code=${this.state.code}`,
        { method: 'POST',
          headers: {
            'Content-Type': 'application/x-www-form-urlencoded', 
            'accept': 'application/json',
          }
        })
      const data = await response.json()
      await this.setState({token: data.access_token})
      await this.getComments()
    } catch (error) {
      console.log('error', error.message)
    }
  }

  getComments = async () => {
    const response = await fetch('https://freesound.org/apiv2/search/combined/?target=rhythm.bpm:120&filter=tag:loop', {
      headers: {
        "Authorization": `Bearer ${this.state.token}`
      }
    })
    const data = await response.json()
    console.log(data)
  }

  render() {
    return (
      <div className="App">
        <Header 
          desktopClicked={this.state.desktopClicked}
          handleDesktopClick={this.handleDesktopClick}
          logOut={this.logOut}
          loggedStatus={this.state.code}/>
        <Route exact path='/desktop-controller' render = {() => {
          return <DesktopDisplayContainer 
            handleDesktopClick={this.handleDesktopClick}
            getComments={this.getComments}/>
        }} />
        <Route exact path='/import' render={() => {
          return <ImportContainer />
        }} />
        <Route exact path='/new-song' render={() => {
          return <NewSongContainer />
        }} />
        <Route exact path='/workout' render={() => {
          return <WorkoutContainer />
        }} />
        <Route exact path='/' render={() => {
          return (
          <div>
            {!this.state.code.length && 
            <a href={`https://freesound.org/apiv2/oauth2/authorize/?client_id=${clientId}&response_type=code`}>Login with freesound</a>}
            {this.state.code.length && 
            <Link to='/desktop-controller'>
               <button onClick={this.handleSubmit}>Get Started!</button>
            </Link>}
          </div>
          )}} />
      </div>
    );
  }
}

export default withRouter((App));