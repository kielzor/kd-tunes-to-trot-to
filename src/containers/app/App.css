.App {
  text-align: center;
  background-color: #2E4756;
  height: 100vh;
  color: #DBC2CF; }

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px; }

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white; }

.App-title {
  font-size: 1.5em; }

.App-intro {
  font-size: large; }

.get-started-button {
  height: 4rem;
  margin-top: 6rem;
  width: 8rem;
  background-color: #3C7A89;
  color: white; }

.a-tag {
  height: 10rem;
  color: #DBC2CF; }
