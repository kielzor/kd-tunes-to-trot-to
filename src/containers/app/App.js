import React, { Component } from 'react'
import './App.css';
import { clientId } from '../../keys'
import Header from '../../components/header/header'
import DesktopDisplayContainer from '../../components/desktopDisplayContainer/DesktopDisplayContainer.js'
import { Route, withRouter, Link } from 'react-router-dom';
import ImportContainer from '../importContainer/ImportContainer'
import { NewSongContainer } from '../../components/newSongContainer/NewSongContainer'
import WorkoutContainer from '../workoutContainer/WorkoutContainer'
import MusicPlayer from '../../containers/musicPlayer/MusicPlayer'
import { fetchUser, fetchAudio } from '../../helpers/Fetch'
import { connect } from 'react-redux'
import { storeJog, storeWarmup, storeSprint, populateWorkout } from '../../actions/index'

export class App extends Component {
  constructor() {
    super()
    this.state = {
      code: ''
    }
  }
  
  componentDidMount = () => {
    let data = window.location.search
    this.setState({
      code: data.split('').splice(6,data.length).join('')})
  }

  logOut = () => {
    this.setState({
      code: ''
    })
  }

  handleSubmit = async () => {
    const token = await fetchUser(this.state.code)
    const jogAudio = await fetchAudio(token, '14826')
    await this.props.addJog(jogAudio)
    const warmupAudio = await fetchAudio(token, '11118')
    await this.props.addWarmup(warmupAudio)
    const sprintAudio = await fetchAudio(token, '14826')
    await this.props.addSprint(sprintAudio)
    if (localStorage.length) {
      const item = JSON.parse(localStorage.getItem('workout'))
      this.props.populateWorkout(item)
    }
  }

  render() {
    return (
      <div className="App">
        <Header 
          desktopClicked={this.state.desktopClicked}
          logOut={this.logOut}
          loggedStatus={this.state.code}/>
        <Route exact path='/desktop-controller' render = {() => {
          return <DesktopDisplayContainer 
            handleDesktopClick={this.handleDesktopClick}/>
        }} />
        <Route exact path='/import' render={() => {
          return <ImportContainer />
        }} />
        <Route exact path='/new-song' render={() => {
          return <NewSongContainer />
        }} />
        <Route exact path='/workout' render={() => {
          return <WorkoutContainer />
        }} />
        <Route exact path='/music-player' render={() => {
          return <MusicPlayer />
        }} />
        <Route exact path='/' render={() => {
          return (
        <div>
          {!this.state.code && 
          <a href={`https://freesound.org/apiv2/oauth2/authorize/?client_id=${clientId}&response_type=code`}>Login with freesound</a>}
          {this.state.code && 
          <Link to='/desktop-controller'>
            <button 
              className='get-started-button'
              onClick={this.handleSubmit}>Get Started!</button>
          </Link>}
        </div>
        )}} />
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  addJog: audioClip => dispatch(storeJog(audioClip)),
  addWarmup: audioClip => dispatch(storeWarmup(audioClip)),
  addSprint: audioClip => dispatch(storeSprint(audioClip)),
	populateWorkout: file => dispatch(populateWorkout(file))
})

export default withRouter(connect(null, mapDispatchToProps)(App));