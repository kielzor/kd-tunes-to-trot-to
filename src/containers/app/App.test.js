import React from 'react'
import { App, mapDispatchToProps } from './App'
import { shallow } from 'enzyme'
import { storeJog, storeWarmup, storeSprint, populateWorkout } from '../../actions/index'

describe('App', () => {
  let wrapper
  let mockAddJog = jest.fn()
  let mockAddWarmup = jest.fn()
  let mockAddSprint = jest.fn()
  let mockPopulateWorkout = jest.fn()

  beforeEach(() => {
    wrapper = shallow (
    <App 
      addJog={mockAddJog}
      addWarmup={mockAddWarmup}
      addSprint={mockAddSprint}
      populateWorkout={mockPopulateWorkout}
    />)})

    window.fetch = jest.fn().mockImplementation(() => 
    Promise.resolve({
      ok: true, 
      json: () => Promise.resolve({ results: '09909' })
    }));

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  describe('logOut', () => {
    it('should reset the state', () => {
      const mockState = { code: 'hello' }
     
      wrapper.setState (mockState)
      wrapper.instance().logOut()

      expect(wrapper.state('code')).toEqual('')
    })
  })

  describe('handleSubmit', () => {
    it('should call mockJog when handleSubmit is called', async () => {
      await wrapper.instance().handleSubmit(10010)

      expect(mockAddJog).toHaveBeenCalled()
    })

    it('should call mockWarmup when handleSubmit is called', async () => {
      await wrapper.instance().handleSubmit(10010)

      expect(mockAddWarmup).toHaveBeenCalled()
    })

    it('should call mockSprint when handleSubmit is called', async () => {
      await wrapper.instance().handleSubmit(10010)

      expect(mockAddSprint).toHaveBeenCalled()
    })
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with storeJog actions when addJog is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = storeJog({ name: 'jog', jogLength: '1:00' })

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addJog({ name: 'jog', jogLength: '1:00' })

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('calls dispatch with storeWarmup actions when addWarmup is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = storeWarmup({ name: 'warmup', warmupLength: '5:00' })

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addWarmup({ name: 'warmup', warmupLength: '5:00' })

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('calls dispatch with storeSprint actions when addSprint is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = storeSprint({ name: 'sprint', sprintLength: '1:00' })

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addSprint({ name: 'sprint', sprintLength: '1:00' })

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('calls dispatch with populateWorkout actions when populateWorkout is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = populateWorkout({ name: 'populate', workout: { name: 'kiel' }})

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.populateWorkout({ name: 'populate', workout: { name: 'kiel' }})

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})