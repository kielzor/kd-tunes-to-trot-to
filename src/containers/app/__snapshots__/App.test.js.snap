// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addJog={[Function]}
    addSprint={[Function]}
    addWarmup={[Function]}
    populateWorkout={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          desktopClicked={undefined}
          logOut={[Function]}
          loggedStatus=""
/>,
        <Route
          exact={true}
          path="/desktop-controller"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/import"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/workout"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/music-player"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/"
          render={[Function]}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "desktopClicked": undefined,
          "logOut": [Function],
          "loggedStatus": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/desktop-controller",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/import",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/workout",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/music-player",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            desktopClicked={undefined}
            logOut={[Function]}
            loggedStatus=""
/>,
          <Route
            exact={true}
            path="/desktop-controller"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/import"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/workout"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/music-player"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "desktopClicked": undefined,
            "logOut": [Function],
            "loggedStatus": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/desktop-controller",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/import",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/workout",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/music-player",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
